src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -637,195 +341,158 @@ static boolean requiresChangeParentUpdateAction(
-   * issues a request to the CTP project defined by the client configuration stored in the {@code
-   * syncOptions} instance of this class to update the specified category with this list of update
-   * actions. If the update request failed due to a {@link ConcurrentModificationException}, the
-   * method recalculates the update actions required for syncing the {@link Category} and reissues
-   * the update request.
-   *
-   * <p>The {@code statistics} instance is updated accordingly to whether the CTP request was
-   * carried out successfully or not. If an exception was thrown on executing the request to CTP,
-   * the optional error callback specified in the {@code syncOptions} is called.
-   *
-   * @param oldCategory the category to update.
-   * @param newCategory the category draft where we get the new data.
-   * @param updateActions the list of update actions to update the category with.
-   * @return a future which contains an empty result after execution of the update.
-   */
   private CompletionStage<Void> updateCategory(
       @Nonnull final Category oldCategory,
       @Nonnull final CategoryDraft newCategory,
       @Nonnull final List<UpdateAction<Category>> updateActions) {
     final String categoryKey = oldCategory.getKey();
     return categoryService
         .updateCategory(oldCategory, updateActions)
-        .handle((updatedCategory, sphereException) -> sphereException)
+        .handle(ImmutablePair::new)
         .thenCompose(
-            sphereException -> {
+            updateResponse -> {
+              final Throwable sphereException = updateResponse.getValue();
+
               if (sphereException != null) {
                 return executeSupplierIfConcurrentModificationException(
                     sphereException,
-                    () -> refetchAndUpdate(oldCategory, newCategory),
+                    () -> fetchAndUpdate(oldCategory, newCategory),
                     () -> {
-                      if (!processedCategoryKeys.contains(categoryKey)) {
-                        handleError(
-                            format(UPDATE_FAILED, categoryKey, sphereException),
-                            sphereException,
-                            oldCategory,
-                            newCategory,
-                            updateActions);
-                        processedCategoryKeys.add(categoryKey);
-                      }
+                      final String errorMessage =
+                          format(UPDATE_FAILED, categoryKey, sphereException.getMessage());
+                      handleError(
+                          errorMessage,
+                          sphereException,
+                          oldCategory,
+                          newCategory,
+                          updateActions,
+                          1);
+
+                      categoryDraftsToUpdateSequentially.remove(newCategory);
```

Comment:
What if `newCategory` is not in `categoryDraftsToUpdateSequentially`?

Reply:
it's doing internally the check/contains and remove:
From the javadoc of remove method.
```
* Removes the key (and its corresponding value) from this map.
     * This method does nothing if the key is not in the map.
```